app/jobs/redownload_register_job.rb

```diff
@@ -0,0 +1,32 @@
+class RedownloadRegisterJob < ApplicationJob
+  queue_as :default
+
+  class FrontendInvalidRegisterError < StandardError; end
+
+  # Redownload all the entries in a register
+  def perform(register)
+    ActiveRecord::Base.transaction do
+      logger.info("Removing existing #{register.name} records from database")
+
+      Record.where(register_id: register.id).delete_all
+      Entry.where(register_id: register.id).delete_all
+      register.root_hash = nil
+      register.save
+
+      logger.info("Updating #{register.name} in database")
+
+      begin
+        RegisterDownloader.download(register)
+      rescue InvalidRegisterError => e
```

Comment:
I don't think it makes sense for this job to rescue from this exception. This is because this exception only gets thrown when the local root hash differs from the remote one, as we have just deleted the local root hash there will not be one so this exception should not throw, please see: https://github.com/openregister/govuk-registers-api-client/blob/76227df2db8887de40ae0882cd931686947e166c/lib/register_client.rb#L131

Reply:
Good point - I'll just remove this rescue